lang/python/CMakeLists.txt

```diff
@@ -149,3 +149,9 @@ add_custom_command(TARGET ${swig_wt_target}
 )
 
 set_target_properties(${swig_wt_target} PROPERTIES OUTPUT_NAME "_wiredtiger")
+
+# Manually changing the suffix on macOS/Darwin builds since DYLD_LIBRARY_PATH
+# doesn't work well for dynamic modules loaded by python.
+if(WT_DARWIN)
+    set_target_properties(${swig_wt_target} PROPERTIES SUFFIX ".so")
+endif()
```

Comment:
I think Autotools generates `.dylib` suffix filenames as well right now, and the below commands needed to be executed  (in a local run or in Evergreen) to avoid dynamic library loading errors when running the Python test suite. 
```
    $ WT_VERSION=$(m4 aclocal/version.m4)
    $ install_name_tool -change /usr/local/lib/libwiredtiger-${WT_VERSION}.dylib `pwd`/.libs/libwiredtiger-${WT_VERSION}.dylib lang/python/_wiredtiger*.so
    $ install_name_tool -change /usr/local/lib/libwiredtiger-${WT_VERSION}.dylib `pwd`/.libs/libwiredtiger-${WT_VERSION}.dylib .libs/wt
```
Not sure if specifying the suffix here for CMake would achieve a similar effect as what the above commands do.

Reply:
I was aware of those lines when making this change. Those lines seem a bit more specific to the autoconf/libtool build, where it needs to rewrite the dynamic loader instructions inside the swig binary to correctly load the WiredTiger library. 
CMake builds its dynamic libraries pretty differently to libtool where we don't need to change the WiredTiger library install name inside our swig binary. By default the shared library path for libwiredtiger within our CMake produced swig binary looks correct:
```
_wiredtiger.dylib:
	@rpath/_wiredtiger.dylib (compatibility version 0.0.0, current version 0.0.0)
	@rpath/libwiredtiger.10.0.1.dylib (compatibility version 0.0.0, current version 10.0.1)
	/usr/local/opt/python@3.9/Frameworks/Python.framework/Versions/3.9/Python (compatibility version 3.9.0, current version 3.9.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.60.1)
```

The changes I added here just ensure CMake outputs a swig binary ending with the `.so` suffix. This gets around an issue in cpython where it doesn't recognize `DYLD_LIBRARY_PATH` (& `dylib` files by extension): https://github.com/python/cpython/blob/main/Python/dynload_shlib.c#L37-L49 (specifically looks for a `.so` file).

Reply:
Thanks for confirming CMake builds of dynamic libraries and the swig binary!